src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -450,6 +456,53 @@ public static Callable<Void> getUploadFileCallable(
       final int parallel,
       final File srcFile,
       final RemoteStoreFileEncryptionMaterial encMat) {
+    return getUploadFileCallable(
+        stage,
+        srcFilePath,
+        metadata,
+        client,
+        session,
+        command,
+        inputStream,
+        sourceFromStream,
+        parallel,
+        srcFile,
+        encMat,
+        null);
+  }
+
+  /**
+   * A callable that can be executed in a separate thread using executor service.
+   *
+   * <p>The callable does compression if needed and upload the result to the table's staging area.
+   *
+   * @param stage information about the stage
+   * @param srcFilePath source file path
+   * @param metadata file metadata
+   * @param client client object used to communicate with c3
+   * @param session session object
+   * @param command command string
+   * @param inputStream null if upload source is file
+   * @param sourceFromStream whether upload source is file or stream
+   * @param parallel number of threads for parallel uploading
+   * @param srcFile source file name
+   * @param encMat not null if encryption is required
+   * @param queryId last executed query id (for forwarding in possible exceptions)
+   * @return a callable that uploading file to the remote store
+   */
+  public static Callable<Void> getUploadFileCallable(
+      final StageInfo stage,
+      final String srcFilePath,
+      final FileMetadata metadata,
+      final SnowflakeStorageClient client,
+      final SFSession session,
+      final String command,
+      final InputStream inputStream,
+      final boolean sourceFromStream,
+      final int parallel,
+      final File srcFile,
+      final RemoteStoreFileEncryptionMaterial encMat,
```

Comment:
minor but I'd prefer full names

Reply:
I also prefer long names but it's named here in such way